packages/headless/src/api/platform-client.ts

```diff
@@ -76,7 +82,15 @@ export class PlatformClient {
           shouldRetry && logger.info('Platform retrying request');
           return shouldRetry;
         },
+      }).catch((error: Error) => {
+        if (error.message !== 'Failed to fetch') {
+          throw error;
+        }
+        return PlatformClientErrorResponse.Disconnected;
```

Comment:
Why are most errors being thrown, while the Disconnected error is treated like a successful response?

Could we instead handle the Disconnected case inside the function that is normally catching thrown errors?

Reply:
Can it be handled along with the AbortError, in the catch at line 92?

Reply:
The `return` here changes the type of the above `const response` from `Response` to `Response | PlatformClientErrorResponse`, allowing us to handle the error outside the promise.

Line 92 doesn't handle the `AbortError`, it handles the Disconnected error.